main/release/releasability/releasability.py

```diff
@@ -0,0 +1,196 @@
+    pass
+
+
+class Releasability:
+    SQS_MAX_POLLED_MESSAGES_AT_A_TIME = 10
+    SQS_POLL_WAIT_TIME = 10
+    FETCH_CHECK_RESULT_TIMEOUT_SECONDS = 60 * 5
+    FETCH_SLEEP_TIME_SECONDS = 2
+
+    ARN_SNS = 'arn:aws:sns'
+    ARN_SQS = 'arn:aws:sqs'
+
+    release_request: ReleaseRequest
+    session: Session
+
+    def __init__(self, release_request):
+        self.release_request = release_request
+        self.session = boto3.Session(region_name=releasability_aws_region)
+        account_id = self._get_aws_account_id()
+        self._define_arn_constants(releasability_aws_region, account_id)
+
+    @Dryable(logging_msg='{function}({args}{kwargs})')
+    def _get_aws_account_id(self) -> str:
+        return boto3.client('sts').get_caller_identity().get('Account')
+
+    def _define_arn_constants(self, aws_region: str, aws_account_id: str):
+        self.TRIGGER_TOPIC_ARN = f"{Releasability.ARN_SNS}:{aws_region}:{aws_account_id}:ReleasabilityTriggerTopic"
+        self.RESULT_TOPIC_ARN = f"{Releasability.ARN_SNS}:{aws_region}:{aws_account_id}:ReleasabilityResultTopic"
+        self.RESULT_QUEUE_ARN = f"{Releasability.ARN_SQS}:{aws_region}:{aws_account_id}:ReleasabilityResultQueue"
+
+    @Dryable(logging_msg='{function}({args}{kwargs})')
+    def start_releasability_checks(self):
+        standardized_version = VersionHelper.as_standardized_version(self.release_request)
+
+        print(f"Starting releasability check: {self.release_request.project}#{standardized_version}")
+
+        correlation_id = str(uuid.uuid4())
+        sns_request = self._build_sns_request(
+            correlation_id=correlation_id,
+            organization=self.release_request.org,
+            project_name=self.release_request.project,
+            branch_name=self.release_request.branch,
+            version=standardized_version,
+            revision=self.release_request.sha,
+            build_number=int(self.release_request.buildnumber)
+        )
+
+        response = self.session.client("sns").publish(
+            TopicArn=self.TRIGGER_TOPIC_ARN,
+            Message=str(sns_request),
+        )
+        print(f"Issued SNS message {response['MessageId']}; the request identifier is {correlation_id}")
+        return correlation_id
+
+    def _build_sns_request(self,
+                           correlation_id: str,
+                           organization: str,
+                           project_name: str,
+                           branch_name: str,
+                           revision: str,
+                           version: str,
+                           build_number: int):
+        sns_request = {
+            'uuid': correlation_id,
+            'responseToARN': self.RESULT_TOPIC_ARN,
+            'repoSlug': f'{organization}/{project_name}',
+            'version': version,
+            'vcsRevision': revision,
+            'artifactoryBuildNumber': build_number,
+            'branchName': branch_name
+        }
+        return sns_request
```

Comment:
What is the benefit of having a function for returning the object?

Reply:
**TLDR;** SRP principle
it centralizes the build of the object with some exotic customizations such as here `repoSlug` which is constructed from two input parameters (we are not in the use case of a 1:1) this methods act as a builder and when later on we have to add/remove some fields we can change it here without having to change other portions of the code.